### DISCUSSION TITLE
Folders Step 1: Category migration to remove bundle relationship

### DISCUSSION THREAD

**Discussion by kuzmany** [mautic/mautic#13019] [Announcements]
Based on https://docs.google.com/document/d/1nJxRDGslOg5aIxoLyK2RwGz1zek9i_DD9L555vnDHN4/edit 

Current situation:
- We have global categories.
- We have categories related to bundles (emails, SMS...).

Before we take the next step based on the document above, we need to migrate the categories and remove the bundle relation. Seems like nobody understand this and nobody use it.

Consequences:

- If we remove the bundle relation, all categories will become global. Does it OK?  This means there might be duplication in some cases.

**Comment by atowne-mautic** [comment]
@kuzmany could there be a migration written for upgrades where categories within a bundle are just merged with existing global categories that have the same alias (and any categories within a bundle are just migrated to be global categories, or whatever we may call them, as you suggest)?
![image](https://github.com/mautic/mautic/assets/46789594/0e433d7a-afac-4d77-8938-2f7815abba62)

**Reply by kuzmany** [reply]
Yes, we can merge it based  on alias. Good point

**Comment by RCheesley** [comment]
I could maybe see where people might have for example preference centres which are based on categories, where this might cause a problem?

Is there some way we can take that into account, as it _could_ impact on what people are/are not subscribed to?

**Comment by kuzmany** [comment]
@RCheesley It should have an impact. The fact that the categories for the preferences center were never deeply integrated. 

But I think, based on some PRs, Acquia could use it (@escopecz ). 

We need to figure out what we're going to do with it. At the moment, I don't have any idea. I need to find time to think about it.

**Reply by escopecz** [reply]
Yes, some of our clients are using categories in the preference center. And I bet some community users as well. Thanks for thinking about this!

**Comment by kuzmany** [comment]
From my point of view, nothing should change.
Currently, contacts are linked to categories in the table called lead_categories.
After we move things over, we will use the same kind of system for other tables related to entities.

```
email_categories
form_categories 
sms_categories 
```

....

After moving, all items should be able to fit into more than one category. My part, the main categories, won't change and shouldn't cause any problems.
The only difference is picking which categories to display in the preferences area. Right now, it shows all the global categories. This seems like a small issue.

**Reply by kuzmany** [reply]
@escopecz can you take one minute? to check it?

**Reply by escopecz** [reply]
Perhaps we could add a new switch option to the category edit page to let users decide whether to "show in preference center"? Same as we do on the segment edit page.
![Screenshot 2024-02-26 at 14 47 57](https://github.com/mautic/mautic/assets/1235442/fd833b08-dc9d-4ae9-ab92-ab782e0919b0)
All current global categories would have this switch on and other will have it off.

**Reply by kuzmany** [reply]
Yes, we can figure it out somehow while building. I believe we now have the basic working concept, and some questions will arise during the process.
